### PR TITLE
docs: Add the vhost-user block test doc

### PR DESCRIPTION
**Pull request by yangzhon**
As #721 describe, we need to document vhost-user-block test like vhost-user-net did before, please help review and comments, thanks.

### CODE REVIEW

File: docs/vhost-user-blk-testing.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# How to test Vhost-user block with SPDK
```
Comment (rbradford) [inline docs/vhost-user-blk-testing.md:1]: Inconsistent with below where you say "vhost-user-blk"

File: docs/vhost-user-blk-testing.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+The purpose of this document is to illustrate how to test vhost-user-blk in cloud-hypervisor with SPDK as the backend.
```
Comment (rbradford) [inline docs/vhost-user-blk-testing.md:3]: ^ here. s/cloud-hypervisor/Cloud Hypervisor/

File: docs/vhost-user-blk-testing.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+- Enable hugepages
+- Install SPDK
+
+Here are some good references for detailing them.
+- spdk
+	* https://spdk.io/doc/
+
+## Test environment
+
+The below test environment is based on ubuntu release, as for other system, please check related document.
```
Comment (rbradford) [inline docs/vhost-user-blk-testing.md:38]: Which Ubuntu release.

File: docs/vhost-user-blk-testing.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+git submodule update --init
+
+# create the build dep
+./scripts/pkgdep.sh
+
+# build spdk
+./configure
+make
+
+# set the SPDk environment
```
Comment (rbradford) [inline docs/vhost-user-blk-testing.md:56]: Strange capitalisation and inconsistent with above.

File: docs/vhost-user-blk-testing.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+sudo HUGEMEM=2048 scripts/setup.sh
+sudo ./app/vhost/vhost -S /var/tmp -s 1024 -m 0x3 &
+
+# create 512M block device
+sudo scripts/rpc.py bdev_malloc_create 512 512 -b Malloc0
+sudo scripts/rpc.py vhost_create_blk_controller --cpumask 0x1 vhost.1 Malloc0
+
+_Launch the VM_
+
+VMs run in client mode. They connect to the socket created by the `dpdkvhostuser` backend.
```
Comment (rbradford) [inline docs/vhost-user-blk-testing.md:68]: Is the backend really called `dpdkvhostuser` ?

Comment (uMartinXu) [reply docs/vhost-user-blk-testing.md:68]: Glad to help. But I am busy with other tasks. So if it is not so urgent, I can help with it. :)

**Comment by uMartinXu** [comment]
I am glad to help if this task is not so urgent. I am busy working on some other tasks.  :)

File: docs/vhost-user-blk-testing.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+# From the test terminal. We need to create one vhost-user block device for the --disk.
+./cloud-hypervisor \
+        --cpus boot=4 \
+        --memory size=1024M,file=/dev/hugepages \
+        --kernel linux/arch/x86/boot/compressed/vmlinux.bin \
+        --cmdline "console=ttyS0 reboot=k panic=1 nomodules i8042.noaux i8042.nomux i8042.nopnp i8042.dumbkbd root=/dev/vda1 iommu=off" \
+        --disk "path=images/test.img" "num_queues=4,queue_size=128,vhost_user=true,socket=/var/tmp/vhost.1,wce=true" \
+        --console off \
+        --serial tty \
+        --seccomp false \
```
Comment (uMartinXu) [inline docs/vhost-user-blk-testing.md:79]: I have to remove --seccomp to start clh

File: docs/vhost-user-blk-testing.md (hunk 1 above)
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:1]: Rob, thanks, i will change this.

File: docs/vhost-user-blk-testing.md (hunk 2 above)
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:3]: Thanks, i will use Cloud Hypervisor to replace cloud-hypervisor.

File: docs/vhost-user-blk-testing.md (hunk 3 above)
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:38]: My Ubuntu release is Ubuntu 16.04.1 LTS, 

From @uMartinXu's test, his release is Ubuntu 18.04.3 LTS.

So,  which release is more suitable here? thanks.

File: docs/vhost-user-blk-testing.md (hunk 5 above)
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:68]: Sorry, this should be "`dpdkvhostuser` in the SPDK backend", thanks.

File: docs/vhost-user-blk-testing.md (hunk 4 above)
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:56]: Rob, thanks, i will change this.

File: docs/vhost-user-blk-testing.md (hunk 6 above)
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:79]: New CLH command will remove this, thanks.

File: docs/vhost-user-blk-testing.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+_Launch the VM_
+
+VMs run in client mode. They connect to the socket created by the `dpdkvhostuser` backend.
+```bash
+# From the test terminal. We need to create one vhost-user block device for the --disk.
+./cloud-hypervisor \
+        --cpus boot=4 \
+        --memory size=1024M,file=/dev/hugepages \
+        --kernel linux/arch/x86/boot/compressed/vmlinux.bin \
+        --cmdline "console=ttyS0 reboot=k panic=1 nomodules i8042.noaux i8042.nomux i8042.nopnp i8042.dumbkbd root=/dev/vda1 iommu=off" \
```
[the remark this answers is not in the document]
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:75]: Sorry, my image is clear-18080-kvm.img, which is different with CLH image. I will change this, thanks.

File: docs/vhost-user-blk-testing.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+├─vda2 253:2    0     1K  0 part
+├─vda5 253:5    0     2G  0 part
+└─vda1 253:1    0   7.8G  0 part /
+
+The vhost-user block device is /dev/vdb
+
+# How to do simple read/write test
+time dd if=/dev/vdb of=/dev/null bs=2M iflag=direct
+time dd of=/dev/vdb if=/dev/zero bs=2M oflag=direct count=256
+
```
[the remark this answers is not in the document]
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:101]: Thanks, let me remove "time" before the dd command, thanks.

File: docs/vhost-user-blk-testing.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+# create the build dep
+./scripts/pkgdep.sh
+
+# build spdk
+./configure
+make
+
+# set the SPDk environment
+sudo sysctl -w vm.nr_hugepages=2
```
[the remark this answers is not in the document]
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:57]: The "hugepages=2" in the host kernel cmdline, so this line can be removed, thanks.

File: docs/vhost-user-blk-testing.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:75]: My lsblk command output like below
lsblk
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
vdb    253:16   0   512M  0 disk
vda    253:0    0   9.8G  0 disk
├─vda2 253:2    0     1K  0 part
├─vda5 253:5    0     2G  0 part
└─vda1 253:1    0   7.8G  0 part /

@uMartinXu , what's your lsblk command output?  If i changed the /dev/vda1 to /dev/sda3, the lsblk command output should be also different. thanks.

File: docs/vhost-user-blk-testing.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+- spdk
+	* https://spdk.io/doc/
+
+## Test environment
+
+The below test environment is based on ubuntu release, as for other system, please check related document.
+The test runs with multiple queue (MQ) support enabled, using 4 queues defined for both SPDK and the virtual machine. Here are the details on how the test can be run.
+
+# Host linux cmdline setting
+Add "default_hugepagesz=1G hugepagesz=1G hugepages=2" into /boot/grub/grub.cfg
```
[the remark this answers is not in the document]
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:42]: All hugepage settings will be put in "### The hugepage setting in host linux", thanks.

File: docs/vhost-user-blk-testing.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (yangzhon) [reply docs/vhost-user-blk-testing.md:75]: I used the clear image to check again, it is /dev/vda3, which is also used in vhost-user-net doc. thanks.

**Comment by yangzhon** [comment]
@rbradford @uMartinXu ,

I have updated the doc for vhost-user-blk test with SPDK, please review again, thanks.

**Comment by yangzhon** [comment]
It's strange that why one doc PR result in below failure?

failures:
    tests::test_memory_overhead

test result: FAILED. 56 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--test integration'

**Comment by sboeuf** [comment]
@uMartinXu please let us know if the documentation is correct now :)

File: docs/vhost-user-blk-testing.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (uMartinXu) [reply docs/vhost-user-blk-testing.md]: that is great. :)

**Comment by uMartinXu** [comment]
@uMartinXu Now, this document is good for me. :)

**Pull request merged**